src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1085,291 +1047,200 @@ internal static string GetExpandedEnvironmentVariable(string name, EnvironmentVa
-            // if $PSHome\Modules IS found (psHomePosition >= 0) - insert <Program Files> location before $PSHome\Modules
-            currentProcessModulePath = AddToPath(currentProcessModulePath, sharedModulePath, indexOfPSHomeModulePath);
-
-            return currentProcessModulePath;
-        }
-
-        /// <summary>
-        /// Checks if $env:PSModulePath is not set and sets it as appropriate. Note - because these
-        /// strings go through the provider, we need to escape any wildcards before passing them
-        /// along.
-        /// </summary>
-        internal static string GetModulePath()
-        {
-            string currentModulePath = GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.Process);
-            return currentModulePath;
+            int indexOfSharedModulePath = IndexOfPath(sharedModulePath);
+            if (indexOfSharedModulePath == -1)
+            {
+                // if we reached this point - always add <Program Files> location to EVT.Process
+                // everything below is the same behaviour as WMF 4 code.
+                // Index of $PSHome\Modules in currentProcessModulePath
+                int indexOfPSHomeModulePath = IndexOfPath(psHomeModulePath);
+
+                // if $PSHome\Modules not found (psHomePosition == -1) - append <Program Files> location to the end;
+                // if $PSHome\Modules IS found (psHomePosition >= 0) - insert <Program Files> location before $PSHome\Modules
+                if (indexOfPSHomeModulePath != -1)
+                {
+                    s_modulePath.Insert(indexOfPSHomeModulePath, sharedModulePath);
+                }
+                else
+                {
+                    s_modulePath.Add(sharedModulePath);
+                }
+            }
         }
 
         /// <summary>
-        /// Checks if $env:PSModulePath is not set and sets it as appropriate. Note - because these
-        /// strings go through the provider, we need to escape any wildcards before passing them
-        /// along.
+        /// Get all standard module paths.
+        /// Note - because these strings go through the provider, we need to escape any wildcards before passing them along.
         /// </summary>
-        private static string SetModulePath()
+        internal static IEnumerable<string> GetModulePath()
         {
+            // From %PSModulePath% environment variable
             string currentModulePath = GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.Process);
-            string allUsersModulePath = PowerShellConfig.Instance.GetModulePath(ConfigScope.AllUsers);
-            string personalModulePath = PowerShellConfig.Instance.GetModulePath(ConfigScope.CurrentUser);
 
-            string newModulePathString = GetModulePath(currentModulePath, allUsersModulePath, personalModulePath);
+            // Although Windows is case-insensetive we can be case-sensetive in the case.
```

Comment:
```suggestion
            // Although Windows is case-insensetive we can be case-sensetive in this case.
```

We should have an explanation as to why we should be case-sensitive here

Reply:
I don't know how say this shortly. Expectation is that nobody on Windows will do a change (and frequently) just for the sake of a char case.